docs/proposal-universal-id.md

```diff
@@ -0,0 +1,351 @@
+  Cylc Flow.
+* Enables multi-user control from the CLI via the UIS without Cylc Flow having
+  to know anything about authorisation.
+* Provides the golden `cylc stop '*'`.
+
+### Bash Completion
+
+Propose a new enhanced (and ridiculously simple) bash completion as a solution
+to the *"I would like to run Cylc commands for my workflow without having to
+type the `REG`"* use cases.
+
+Reference Implementation:
+
+> **Note:** This is psudocode to illustrate where the bash autocompletion
+  would get its suggestions from.
+
+```
+# 𝄄 = the users cursor
+
+# autocomplete command name
+$ cylc 𝄄
+echo $CMD_LIST  # can be built during pip install (would pick up plugins)
+
+# autocomplete flow (REG)
+$ cylc trigger 𝄄
+if [[ $1 == cylc && trigger == $FLOW_CMD_LIST ]]; then
+    # the CLI is expecting a REG
+    if [[ -d '.service' ]]; then
+        # we are in the run dir
+        echo $(basename "$PWD")
+    elif [[ -d 'flow.cylc' ]]; then
+        # we are in the src dir
+        echo $(basename "$PWD")
+    else
+        # provide a list of workflows
+        cylc scan --state=all --format=name
+    fi
+fi
+
+# autocomplete cycle
+$ cylc trigger foo// 𝄄
+echo ~/cylc-run/foo/log/job
+
+# autocomplete task
+$ cylc trigger foo//123 𝄄
+echo ~/cylc-run/foo/log/job/123 | reverse  # newest cycle points first
+
+# works with globs too
+$ cylc trigger foo//1* 𝄄
+echo ~/cylc-run/foo/log/job/1* | reverse  # newest cycle points first
+
+# autocomplete job
+$ cylc trigger foo//123/foo 𝄄
+echo ~/cylc-run/foo/log/job/123/foo
+
+# autocomplete flow (REG) for another user
+$ cylc trigger ~user/ 𝄄
+cylc scan --state=all --format=name --dir=~user/cylc-run
+# note the --dir arg would be a two line change to add
+```
+
+Benefits:
+
+* Easy to issue commands form the run dir, press tab to autocomplete the `REG`.
+* Also easy to issue commands from the src dir, again just press tab.
+* Provides autocompletion for cycles/tasks/jobs in the n=0 window and
+  from historical execution.
+* Cylc commands look the same wherever they are run from making it clear
+  what the command is doing.
+  All operations are explicit with the name of the workflow
+  appearing in the command before issue.
+
+Motivations:
+
+* We need to rewrite the completions anyway.
+* Typing cycle points is hard.
+
+### CLI
+
+Propose the following interface for all commands which take `REG` as an
+argument:
+
+```
+cylc sub-commands ID [ID ...]
+```
+
+E.G:
+
+```console
+$ cylc trigger flow//cycle
+$ cylc trigger flow//cycle/*
+$ cylc trigger flow//cycle flow2//cycle2 flow3//cycle3
+```
+
+If the first argument stops at the `//` and the following arguments start
+with `//` then they should be implicitly joined e.g:
+
+```console
+# trigger cycles1, cycle2 and cycle3 in flow
+$ cylc trigger flow// //cycle1 //cycle2 //cycle3
+```
+
+The `//` serving as the marker to the bash completion to glob cycle points.
+
+## Back Compat
+
+We should be able to support the old legacy interface for the time being.
```

Comment:
I think I'd be happy to drop the legacy interface for most commands and only worry about those we think may have been used within scripts/workflows (e.g. broadcast).